src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -225,26 +193,39 @@ private void LogMessage(string message, Exception e = null, string database = nu
                     {
                         if (rawRecord == null)
                         {
-                            LogMessage($"Can't analyze the stats of database the {database}, because the database record is null.", database: database);
+                            _observerLogger.Log($"Can't analyze the stats of database the {database}, because the database record is null.", iteration: _iteration, database: database);
                             continue;
                         }
 
                         if (rawRecord.IsSharded)
                         {
-                            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal, "We need to refactor this class and reuse most of UpdateDatabaseTopology");
                             var databaseName = rawRecord.DatabaseName;
                             var topology = rawRecord.Sharding.Orchestrator.Topology;
-                            var updateReason = UpdateOrchestratorTopology(databaseName, newStats, prevStats, topology, clusterTopology);
+                            var state = new DatabaseObservationState
+                            {
+                                Name = databaseName,
+                                DatabaseTopology = topology,
+                                ClusterTopology = clusterTopology,
+                                Current = newStats,
+                                Previous = prevStats,
+                                RawDatabase = rawRecord,
+                                LastIndexModification = etag,
+                                ObserverIteration = _iteration
+                            };
+
+                            var updateReason = _orchestratorTopologyUpdater.Update(context, state, ref deletions);
+                            deletions = null; // database deletions are irrelevant in orchestrator topology changes
```

Comment:
If I understand this correctly, this `deletions` are sharded between different databases... setting this to `null` might clear thing that aren't relevant for this database, no?

Reply:
If so, I would suggest to make this as an `out` param and add to the global list if needed.